lib/common/options.c

```diff
@@ -1051,48 +1064,57 @@ pcmk__format_option_metadata(const char *name, const char *desc_short,
         // The standard requires a parameter type
         CRM_ASSERT(opt_type != NULL);
 
-        pcmk__g_strcat(s,
-                       "    <" PCMK_XE_PARAMETER " "
-                               PCMK_XA_NAME "=\"", opt_name, "\">\n", NULL);
+        parameter = create_xml_node(parameters, PCMK_XE_PARAMETER);
+        if (parameter == NULL) {
+            goto done;
+        }
+
+        crm_xml_add(parameter, PCMK_XA_NAME, opt_name);
 
-        add_desc(s, PCMK_XE_LONGDESC, opt_desc_long, "      ");
-        add_desc(s, PCMK_XE_SHORTDESC, opt_desc_short, "      ");
+        if (add_desc(parameter, PCMK_XE_LONGDESC,
+                     opt_desc_long) != pcmk_rc_ok) {
```

Comment:
You checked NULL before doing add_desc() above ... if you do it in add_desc(), we don't have to worry about it

Reply:
That's because the standard doesn't require longdesc and shortdesc at the top-level. They're required only for parameters.

So at the top-level, we add the description only if the description exists. At the parameter level, we assert that at least one of the descriptions is non-NULL. If only one is non-NULL, we use it for both.

FYI, in 72a254a8 (#3338), I assert non-NULL at the top level too, even though the standard doesn't require it. It's easier that way. We're in control of the metadata and we can guarantee that the longdesc and shortdesc are both present at the top level.

Reply:
yep that's fine